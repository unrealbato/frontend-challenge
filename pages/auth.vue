<template>
  <v-row class="height-100 flex-column flex-md-row ma-0">
    <v-col class="teal flex-grow-0 flex-md-grow-1">
      <h1 class="ml-5 font-weight-bold text--center">PushBot</h1>
    </v-col>

    <!--User Login Form-->
    <v-col class="d-flex align-center justify-center">
      <v-form
        ref="loginForm"
        class="width-75"
        lazy-validation
        @submit.prevent="onSubmit()"
      >
        <v-row class="flex-column">
          <v-col>
            <h2 class="title">Log in to your account</h2>
          </v-col>

          <v-col>
            <v-text-field
              v-model="email"
              dense
              filled
              label="Your Email Address"
              prepend-inner-icon="mdi-email-outline"
              type="email"
              required
              :rules="emailCheck"
            ></v-text-field>
            <v-text-field
              v-model="password"
              dense
              filled
              label="Your Password"
              prepend-inner-icon="mdi-lock-outline"
              type="password"
              required
              :rules="passwordCheck"
            ></v-text-field>
          </v-col>

          <v-col>
            <v-btn depressed small text class="d-block"
              >forget your password?</v-btn
            >
            <v-btn depressed small text class="d-block"
              >Don't have an account?</v-btn
            >
          </v-col>

          <v-col>
            <v-btn type="submit" class="" width="100%">LOGIN</v-btn>
          </v-col>

          <v-col class="d-flex">
            <v-btn tile class="red flex-grow-1">GOOGLE</v-btn>
            <v-btn tile class="black flex-grow-1">GITHUB</v-btn>
          </v-col>
        </v-row>
      </v-form>
    </v-col>
  </v-row>
</template>

<script>
export default {
  name: 'Auth',
  components: {},
  layout: 'auth',
  auth: 'guest',
  head() {
    return {
      title: 'User Login',
      meta: [
        {
          hid: 'description',
          name: 'description',
          content: 'PushBot User Login Page'
        }
      ]
    }
  },
  data() {
    return {
      email: null,
      emailCheck: [
        (v) => !!v || 'E-mail is required',
        (v) => /.+@.+\..+/.test(v) || 'E-mail must be valid'
      ],
      password: null,
      passwordCheck: [(v) => !!v || 'Password is required'],
      trigger: false,
      message: null,
      color: null
    }
  },

  created() {},

  methods: {
    async onSubmit() {
      if (this.$refs.loginForm.validate()) {
        const credential = {
          email: this.email,
          password: this.password
        }
        try {
          await this.$auth.loginWith('local', { data: credential })
          this.$store.commit('activeSnackbar', ['green', 'Successfully login'])
        } catch (err) {
          this.$store.commit('activeSnackbar', ['red', err.response.data.msg])
        }
      }
    }
  }
}
</script>

<style scoped>
.width-75 {
  width: 75%;
}

@media only screen and (max-width: 991px) {
  .text--center {
    text-align: center;
    margin-left: 0 !important;
  }
}
</style>
