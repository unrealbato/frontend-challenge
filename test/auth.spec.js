import { mount, createLocalVue } from '@vue/test-utils'
import vuetify from 'vuetify'
import Vue from 'vue'
import auth from '@/pages/auth.vue'

function createConfig() {
  const localVue = createLocalVue()
  const propsData = {}
  const mocks = {
    // Vue Auth
    $auth: {
      check: () => false
    },
    // Vuex
    $store: {
      state: {},
      commit: () => {}
    }
  }
  return Object.assign({ mocks, propsData, localVue })
}

describe('Auth Page', () => {
  const config = createConfig()
  let wrapper

  beforeEach(() => {
    Vue.use(vuetify)
    wrapper = mount(auth, config)
  })

  it('HTML Render The Same', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
