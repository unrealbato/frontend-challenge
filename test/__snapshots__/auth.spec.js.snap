// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page HTML Render The Same 1`] = `"<div class=\\"row height-100 flex-column flex-md-row ma-0\\"><div class=\\"teal flex-grow-0 flex-md-grow-1 col\\"><h1 class=\\"ml-5 font-weight-bold text--center\\">PushBot</h1></div> <div class=\\"d-flex align-center justify-center col\\"><form novalidate=\\"novalidate\\" class=\\"v-form width-75\\"><div class=\\"row flex-column\\"><div class=\\"col\\"><h2 class=\\"title\\">Log in to your account</h2></div> <div class=\\"col\\"><div class=\\"v-input v-input--dense theme--light v-text-field v-text-field--filled v-text-field--enclosed\\"><div class=\\"v-input__control\\"><div class=\\"v-input__slot\\"><div class=\\"v-input__prepend-inner\\"><div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-email-outline theme--light\\"></i></div></div><div class=\\"v-text-field__slot\\"><label for=\\"input-3\\" class=\\"v-label theme--light\\" style=\\"right: 0px; position: absolute;\\">Your Email Address</label><input required=\\"required\\" id=\\"input-3\\" type=\\"email\\"></div></div><div class=\\"v-text-field__details\\"><div class=\\"v-messages theme--light\\"><span name=\\"message-transition\\" tag=\\"div\\" class=\\"v-messages__wrapper\\"></span></div></div></div></div> <div class=\\"v-input v-input--dense theme--light v-text-field v-text-field--filled v-text-field--enclosed\\"><div class=\\"v-input__control\\"><div class=\\"v-input__slot\\"><div class=\\"v-input__prepend-inner\\"><div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-lock-outline theme--light\\"></i></div></div><div class=\\"v-text-field__slot\\"><label for=\\"input-7\\" class=\\"v-label theme--light\\" style=\\"right: 0px; position: absolute;\\">Your Password</label><input required=\\"required\\" id=\\"input-7\\" type=\\"password\\"></div></div><div class=\\"v-text-field__details\\"><div class=\\"v-messages theme--light\\"><span name=\\"message-transition\\" tag=\\"div\\" class=\\"v-messages__wrapper\\"></span></div></div></div></div></div> <div class=\\"col\\"><button type=\\"button\\" class=\\"d-block v-btn v-btn--depressed v-btn--flat v-btn--text theme--light v-size--small\\"><span class=\\"v-btn__content\\">forget your password?</span></button> <button type=\\"button\\" class=\\"d-block v-btn v-btn--depressed v-btn--flat v-btn--text theme--light v-size--small\\"><span class=\\"v-btn__content\\">Don't have an account?</span></button></div> <div class=\\"col\\"><button type=\\"submit\\" class=\\"v-btn v-btn--contained theme--light v-size--default\\" style=\\"width: 100%;\\"><span class=\\"v-btn__content\\">LOGIN</span></button></div> <div class=\\"d-flex col\\"><button type=\\"button\\" class=\\"red flex-grow-1 v-btn v-btn--contained v-btn--tile theme--light v-size--default\\"><span class=\\"v-btn__content\\">GOOGLE</span></button> <button type=\\"button\\" class=\\"black flex-grow-1 v-btn v-btn--contained v-btn--tile theme--light v-size--default\\"><span class=\\"v-btn__content\\">GITHUB</span></button></div></div></form></div></div>"`;
